boards: beagle: beaglev_fire: Specify the different targets supported for beaglev_fire and fix build instructions

Ayush1325 opened the pull request:
- The rendered documents should be different for the different zephyr targets on beaglev fire
- Also fix the board name in build example
- Fixes: https://github.com/zephyrproject-rtos/zephyr/issues/79483
- doc preview: https://builds.zephyrproject.io/zephyr/pr/79511/docs/

con-pax commented:
Hey @Ayush1325, thanks for the PR. is it not overkill having 3 seperate docs for the one board? would it not be better to describe the variants in a single doc?

Ayush1325 commented:
> Hey @Ayush1325, thanks for the PR. is it not overkill having 3 seperate docs for the one board? would it not be better to describe the variants in a single doc?

Well, for a board like beagleplay, I would prefer to have separate docs for each target. Having separate docs allows for providing specific instructions for building, debugging, peripheral numbering, etc, more clearly. But I am not familiar enough with the difference between the capabilities of e15 and u54 to say if they are different enough to warrant separate pages. I am just trying to facilitate the future target specific doc improvement.

I would be okay with whatever upstream Zephyr prefers to be honest. It's just important to have docs that can clearly explain that 3 different targets can be used for beaglev-fire.

Ayush1325 committed:
boards: beagle: beaglev_fire: Fix building example

- Fix the board name in build example
- Add link to datasheet
- Specify the different targets supported for beaglev_fire

Signed-off-by: Ayush Singh <ayush@beagleboard.org>

Ayush1325 commented:
@con-pax I now have a single page for beaglev-fire and just added the targets and fixed the build example.

con-pax reviewed (dismissed):
LGTM

mmahadevan108 commented:
@gbcstott, this is the fix to the documentation per your issue. Can you please review.

kartben commented on boards/beagle/beaglev_fire/doc/index.rst:
```diff
@@ -22,11 +22,18 @@ hobbyists, and researchers to explore and experiment with RISC-V technology.
 Building
 ========
 
+BeagleV-Fire supports 3 board targets in Zephyr:
+1. beaglev_fire/polarfire/e51
+2. beaglev_fire/polarfire/u54
+3. beaglev_fire/polarfire/u54/smp
```
```suggestion
BeagleV-Fire supports 3 board targets in Zephyr:

1. ``beaglev_fire/polarfire/e51``
2. ``beaglev_fire/polarfire/u54``
3. ``beaglev_fire/polarfire/u54/smp``
```

kartben replied:
this hasn't been applied properly, blank line still missing
<img width="749" alt="image" src="https://github.com/user-attachments/assets/51806fb2-5413-48f2-8946-b693fe626104">

kartben commented on boards/beagle/beaglev_fire/doc/index.rst:
```diff
@@ -22,11 +22,18 @@ hobbyists, and researchers to explore and experiment with RISC-V technology.
+More information regarding the differences between the cores can be found in the [datasheet](https://ww1.microchip.com/downloads/aemDocuments/documents/FPGA/ProductDocuments/ProductBrief/PolarFire-SoC-Product-Overview-60001656.pdf)
+
 Applications for the ``beaglev_fire`` board configuration can be built as usual:
 
 .. zephyr-app-commands::
    :zephyr-app: samples/hello_world
-   :board: beaglev_fire
+   :board: beaglev_fire/polarfire/{e51,u54,u54/smp}
    :goals: build
```
I would recommend to keep the example command one that can be copy-pasted and "just works". So just show one example, e.g. for E51, and change to "Applications can be built as usual, for example when targeting the E51 core:"

gbcstott commented:
I looked at the doc preview for this change and it looks great. Thanks.

kartben commented on boards/beagle/beaglev_fire/doc/index.rst:
```diff
@@ -22,11 +22,18 @@ hobbyists, and researchers to explore and experiment with RISC-V technology.
+
+More information regarding the differences between the cores can be found in the [datasheet](https://ww1.microchip.com/downloads/aemDocuments/documents/FPGA/ProductDocuments/ProductBrief/PolarFire-SoC-Product-Overview-60001656.pdf)
+
+Here is an example of building ``samples/hello_world`` for E51 core.
```
```suggestion
Here is an example of building :zephyr:code-sample:`hello_world` for E51 core.
```

kartben commented on boards/beagle/beaglev_fire/doc/index.rst:
```diff
@@ -22,11 +22,19 @@ hobbyists, and researchers to explore and experiment with RISC-V technology.
+2. ``beaglev_fire/polarfire/u54``
+3. ``beaglev_fire/polarfire/u54/smp``
+
+More information regarding the differences between the cores can be found in the [datasheet](https://ww1.microchip.com/downloads/aemDocuments/documents/FPGA/ProductDocuments/ProductBrief/PolarFire-SoC-Product-Overview-60001656.pdf)
```
apologies that I missed it but this is not how you create hyperlinks in ReST.
https://docs.zephyrproject.org/latest/contribute/documentation/guidelines.html#cross-referencing-external-content

Ayush1325 commented:
> @Ayush1325 can you describe the process of loading "hello world" without using OpenOCD? Is there a way to enable `west flash`?

I do not have much idea since I do not have a beaglev-fire. But I have posted the link to this on Discord, so hopefully, someone can comment.

Ayush1325 commented:
Closing in favor of https://github.com/zephyrproject-rtos/zephyr/pull/80656
